resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/File.java

```diff
@@ -371,6 +391,32 @@ public void setContents(InputStream content, int updateFlags, IProgressMonitor m
 			subMonitor.done();
 		}
 	}
+	@Override
+	public void setContents(byte[] content, int updateFlags, IProgressMonitor monitor) throws CoreException {
+		String message = NLS.bind(Messages.resources_settingContents, getFullPath());
+		SubMonitor subMonitor = SubMonitor.convert(monitor, message, 100);
+		try {
+			if (workspace.shouldValidate)
+				workspace.validateSave(this);
+			final ISchedulingRule rule = workspace.getRuleFactory().modifyRule(this);
+			SubMonitor newChild = subMonitor.newChild(1);
```

Comment:
Inline?

Reply:
I  don't see the benefit over consistency with the other code around.